lib/spack/spack/cmd/ci.py

```diff
@@ -793,8 +793,13 @@ def ci_verify_versions(args):
 
     failed_version = False
     for pkg_name, path in pkgs:
-        spec = spack.spec.Spec(pkg_name)
-        pkg = spack.repo.PATH.get_pkg_class(spec.name)(spec)
+        try:
+            spec = spack.spec.Spec(pkg_name)
+            pkg = spack.repo.PATH.get_pkg_class(spec.name)(spec)
+
+        # Skip removed packages that are no longer in the repository
+        except spack.repo.UnknownPackageError:
+            continue
```

Comment:
The loop should just know beforehand what is removed.

Can you change this to use `spack.repo.diff_packages()`?  It gives you added and removed packages.  It's used by `spack pkg removed`, `spack pkg changed`, `spack pkg added`, etc. Look at that and see to what extent it makes sense to consolidate logic with `spack.util.git`.

Reply:
Updated this to use logic from `spack.repo`. 

There's a bit of work to be done simplifying the logic between `spack.repo`, `spack.ci`, and `spack.util.git`, that I'll do in a follow up PR to this one since that work is outside of this scope and might be more controversial.